### PR TITLE
Dilpreet | Test | Prod - Select page dropdown is not closing when clicking within the "Add Role" pop up on permission page

### PR DESCRIPTION
**Pull request by dilpreetsinghofficial**
https://app.clickup.com/t/2e5rnb4

### CODE REVIEW

**Review by the-prateek-mishra — APPROVED** [review]

File: apps/web-giddh/src/app/permissions/components/list/permission.list.component.ts [hunk 1]
```diff
@@ -69,7 +69,12 @@ export class PermissionListComponent implements OnInit, AfterViewInit, OnDestroy
         // Getting roles every time user refresh page
         this.store.dispatch(this.permissionActions.GetRoles());
         this.store.dispatch(this.permissionActions.RemoveNewlyCreatedRoleFromStore());
-        this.store.pipe(select(p => p.permission.roles), takeUntil(this.destroyed$)).subscribe((roles: IRoleCommonResponseAndRequest[]) => this.allRoles = roles);
+        this.store.pipe(select(p => p.permission.roles), takeUntil(this.destroyed$)).subscribe((roles: IRoleCommonResponseAndRequest[]) => {
+            this.allRoles = roles
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/permissions/components/list/permission.list.component.ts]:
```suggestion
            this.allRoles = roles;
```

File: apps/web-giddh/src/app/permissions/components/list/permission.list.component.ts [hunk 2]
```diff
@@ -69,7 +69,12 @@ export class PermissionListComponent implements OnInit, AfterViewInit, OnDestroy
...
+            if (this.allRoles.length > 0) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/permissions/components/list/permission.list.component.ts]:
```suggestion
            if (this.allRoles?.length > 0) {
```

File: apps/web-giddh/src/app/permissions/components/model/permission.model.component.html [hunk 3]
```diff
@@ -33,7 +33,15 @@ <h3>{{ localeData?.add_role }}</h3>
...
                 <div class="form-group select-page position-relative" *ngIf="newRoleObj.isFresh">
-                    <div class="btn-group btn-block" dropdown (onShown)="onDDShown()" (onHidden)="onDDHidden()">
+                    <div
+                        class="btn-group btn-block"
+                        dropdown
+                        (onShown)="onDDShown()"
+                        (onHidden)="onDDHidden()"
+                        #permissionDropDown="bs-dropdown"
+                        [attachOutsideOnClick]="true"
+                        (clickOutside)="permissionDropDown.hide()"
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/permissions/components/model/permission.model.component.html]:
```suggestion
                        (clickOutside)="permissionDropDown?.hide()"
```

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

**Comment by dilpreetsinghofficial** [comment]
Reviewed

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**